component/registry.go

```diff
@@ -28,16 +29,47 @@ var (
 	parsedNames = map[string]parsedName{}
 )
 
+// ModuleSystem is a mechanism responsible for allowing components to create other components, or
+// flow systems.
+type ModuleSystem interface {
+	// NewModuleDelegate creates a new, un-started ModuleDelegate.
+	NewModuleDelegate(id string) ModuleDelegate
+}
+
+// ModuleDelegate is a controller for running components within a Module.
+type ModuleDelegate interface {
```

Comment:
Same as the above with making the name more specific to what it is doing, also open to using a different word than `Delegate`